tests/StubsTest.php

```diff
@@ -377,15 +379,50 @@ private static function getDuplicatedFunctions(array $filtered): array
         $duplicatedFunctions = array_filter($filtered, function (PHPFunction $value, int|string $key) {
             if (str_contains($key, 'duplicated')) {
                 $duplicatesOfFunction = self::getAllDuplicatesOfFunction($value->name);
-                $sinceVersionOfInitialFunction = Utils::getDeclaredSinceVersion(
+                $functionVersions[] = Utils::getAvailableInVersions(
                     PhpStormStubsSingleton::getPhpStormStubs()->getFunctions()[$value->name]);
-                $sinceVersionsOfDuplicatedFunctions = array_map(function (PHPFunction $function) {
-                    return Utils::getDeclaredSinceVersion($function);
+                $functionVersions[] = array_map(function (PHPFunction $function) {
+                    return Utils::getAvailableInVersions($function);
                 }, $duplicatesOfFunction);
-                return !empty(array_search($sinceVersionOfInitialFunction, $sinceVersionsOfDuplicatedFunctions));
+                $twoDimensionalArray = [];
+                self::convertArrayToTwoDirectional($functionVersions, $twoDimensionalArray);
+                $hasDuplicates = false;
+                while (($next = array_pop($twoDimensionalArray)) !== null && !empty($twoDimensionalArray)) {
```

Comment:
Will the following work exactly the same?
```
$hasDuplicates = false;
$current = array_pop($array);
while ($next = array_pop($array)) {
    if (!empty(array_intersect($current, $next))) {
        $hasDuplicates = true;
    }
    $current = $next;
}
```

Reply:
Seems this code will not work if we have an original function and 2 duplicates and want to find an intersection of the original (1st) function with 3rd function as code above finds intersection only of 2 adjacent elements but it's required to jump over middle elements. like
`#[PhpStormStubsElementAvailable(from: '5.6', to: '7.3')]`
`#[PhpStormStubsElementAvailable(from: '7.4', to: '8.0')]`
`#[PhpStormStubsElementAvailable(from: '7.2', to: '7.3')]`
In such cases first and the last functions would be duplicates.

Reply:
Good catch! It should be:
```
$hasDuplicates = false;
$current = array_pop($array);
while ($next = array_pop($array)) {
    if (!empty(array_intersect($current, $next))) {
        $hasDuplicates = true;
    }
    $current = array_merge($current, $next);
}
```

Reply:
Hm, `array_merge` will do the trick I guess. Thank you for the suggestion!